### DISCUSSION TITLE
Incorrect _id field from relationship picker resulting in Fetch Row failing

### DISCUSSION THREAD

**Discussion by head-baker** [Budibase/budibase#14372] [Help]
**Checklist**
- [x] I have searched budibase discussions and github issues to check if my issue already exists

**Hosting**
  - Self
  - Method: docker compose
  - Budibase Version: 2.29.28
  - App Version: 2.29.28
  - Tenant ID: N/A using docker compose locally

**Describe the bug**
Unable to use Fetch row when using the value of a relationship picker. The relationship picker yields the value `%5B&#x27;Test%20Customer&#x27;%5D`, which fails when using Fetch Row. The actual ID field is `%5B'1274fdd4-c87c-4b75-b348-e894ae6b1817'%5D` and works when using Fetch Row - I obtained the real ID by accessing {{ Saved Row }}._id.

**To Reproduce**
1. Create two related tables in a postgres db
2. Set up relationships
3. Use a relationship picker and on field change try to use Fetch Row based on the pickers value
4. From here, it should fail to fetch the row as the value from the picker won't have the correct _id field

**Expected behavior**
When I get {{ Field Value }}, I would expect for a relationship picker this should yield the correct _id field that can be used in Fetch Row as indicated in this document here - https://docs.budibase.com/docs/filter-by-relationships.
  
**App Export**
If possible - please attach an export of your budibase application for debugging/reproduction purposes.

**Desktop (please complete the following information):**
 - OS: macOS Monterey 12.5.1
 - Browser: Chrome
 - Version: 125.0.6422.78 

**Additional context**
The goal here is to update another field in a form based on data fetched from a row, so I reallt need this to work to enable this functionality.

**Comment by linear** [comment]
<p><a href="https://linear.app/budibase/issue/BUDI-8541/incorrect-id-field-from-relationship-picker-resulting-in-fetch-row">BUDI-8541 Incorrect _id field from relationship picker resulting in Fetch Row failing</a></p>

**Comment by andz-bb** [comment]
hey @head-baker, which column is selected as the primary key for the relationship? If the ID field is selected I would expect the Relationship Picker to use that as the field value.

**Comment by head-baker** [comment]
The tables are related on a `name` field, so that's what I am using for the relationship. I would be happy to change it, but in my schema I have no `_id` field as this is an internal Budibase field right? If I look to edit the relationship there isn't an option to base if on the `_id` field

**Comment by ConorWebb96** [comment]
Hey @head-baker,

The `_id` is an encoded version of your primary key. So if you were to change your primary key values in anyway. The encoded `_id` will also change along side this.

I hope this makes sense.

**Comment by head-baker** [comment]
Thanks for the prompt response again team. How is it possible that I can get a different value for `{{ Saved Row }}._id` when saving a row to a table from a form and `{{.Field Value }}` from the relationship picker? What am I missing here.

Even if I manually go to the row in the table and copy the `_id` field I get a different value than `{{ Saved Row }}._id`, but the `{{ Saved Row }}._id` field is the only one that actually works with the Fetch Row action
